pkg/api/v20170701/validate.go

```diff
@@ -157,9 +161,18 @@ func (a *Properties) Validate() error {
 	if e := validateUniqueProfileNames(a.AgentPoolProfiles); e != nil {
 		return e
 	}
+
 	if a.OrchestratorProfile.OrchestratorType == Kubernetes {
-		if a.ServicePrincipalProfile == nil {
```

Comment:
If I understand correctly serviceprincipal in the public API model is not set in that case. Only the internal/in memory API model is set to "msi".